.github/workflows/build-and-test-pipeline.yml

```diff
@@ -0,0 +1,18 @@
+name: Continuous Integration
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+      - 'documentation/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'documentation/**'
+jobs: 
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: docker build --target test --tag todo-app:test .
+    - run: docker run -e API_KEY=${{ secrets.TRELLO_API_KEY }} -e API_TOKEN=${{ secrets.TRELLO_API_TOKEN }} -e ORGANISATION_ID=${{ secrets.ORGANISATION_ID }} -e SECRET_KEY=${{ secrets.SECRET_KEY }} todo-app:test
```

Comment:
Nicer to run the tests in two parts for a few reasons
```
- run: docker run todo-app:test tests
- run: docker run [-e  API_KEY=etc...] todo-app:test e2e_tests 
```

- Your integration tests load the .env.test file, clobbering the real env vars. You are implicitly relying on the e2e tests running first because of alphabetical ordering. Not super robust.
- You should minimise what has access to real credentials in case something isn't being mocked correctly
- You can see at a glance if e2e or unit tests are failing or slow